### ISSUE TITLE
Matlab vs JASP discrepancy?

### ISSUE THREAD

**Issue by ojr23** [jasp-stats/jasp-desktop#2431]
## Template for bug reports

* JASP version:
* OS name and version:
* Analysis:
* Steps to reproduce: 

## Template for enhancement requests
 
* Enhancement:
* Purpose:

**Comment by ojr23** [comment]
JASP version: 0.8.6
OS name and version: OSX El Capitain 10.11.6
Analysis: RMANOVA
Steps to reproduce: see attached zipped JASP file
[JASPexample.jasp.zip](https://github.com/jasp-stats/jasp-desktop/files/1830445/JASPexample.jasp.zip)

There is a discrepancy between matlab and JASP (see images below), when the between subject factor has more than 2 levels.

I *think* based on discussions it is probably due to differences in the sums of squares type and plan to check properly when I have a bit more time. Am pasting it here on request of @AlexanderLyNL on twitter so you can check it out. I havent been able to find out (at a v. cursory glance) what Sum Sq JASP uses, so if that is the issue it might be helpful to have that stated somewhere a little more prominent.


![1](https://user-images.githubusercontent.com/5824361/37671481-b73d8d22-2c63-11e8-83fc-d72c99611bf1.jpg)
![2](https://user-images.githubusercontent.com/5824361/37671483-b7620f58-2c63-11e8-9d6e-c183f499709e.jpg)

**Comment by EJWagenmakers** [comment]
@JohnnyDoorn or @Kucharssim ?

**Comment by JohnnyDoorn** [comment]
Hi Ojr23,
Thanks for sharing your data and jasp file, I will look into this discrepancy. The sum of squares type that JASP uses in your example is III, as is stated in the note under the tables. Are you sure that your Matlab code also uses this same type? If so, it might be an issue with how the contrasts are specified in the ANOVA, but I will investigate.
Kind regards,
Johnny

**Comment by ojr23** [comment]
Great! Thanks!

Of course it says the sums of squares under the table! I think Ive just
looked at so many of them that Ive become conditioned to ignore it!

I *think* ranova in matlab also uses type III as a default, but I have
actually been struggling to confirm this in the documentation. Im planning
to look into this properly later in the week...

**Comment by akashrajkn** [comment]
Matlab uses type 3 by default for `anova` (https://nl.mathworks.com/help/stats/compactlinearmodel.anova.html#bszh780-1-sstype)

**Comment by ojr23** [comment]
I suspect that is also true for ranova (which is a different function), but
I havent been able to find it written anywhere. Unlike anova, I dont think
you can specify the sum sq type for ranova (?).

If they are both type 3 then there must be something else going on. There
are a few empty cells in the data, so it could be to do with how missing
data is dealt with?

**Comment by JohnnyDoorn** [comment]
Hi Ojr,
I can reproduce the results of JASP in R (both with aov and ezANOVA), when the rows of the data are removed that contain missing values. From the degrees of freedom in your Matlab output it looks like Matlab does the same, but maybe I'm missing something. Can you share your Matlab code? It might be the case that Matlab handles missing data differently. I cannot seem to reproduce the Matlab results at the moment, neither by changing the SS type nor by handling missing data differently (replacing missing cells with the column means).
Kind regards
Johnny

**Comment by ojr23** [comment]
I was using this function:
https://uk.mathworks.com/matlabcentral/fileexchange/64980-simple-rm-mixed-anova-for-any-design

...with the following code:

datamat=horzcat(propmedhigh_NegOnly,propmedhigh_PosOnly) % two columns of
RTs (for the 4 column version i added the other two columns)
between_factors=triangle1circle0_NegOnly % a column of 1s and 0s
representing groups
tbl = simple_mixed_anova(datamat, between_factors, {'Time'},{'spreadsheet'})

**Comment by JohnnyDoorn** [comment]
Hi Ojr,
Unfortunately I do not have access to Matlab, and the free alternative (Octave) is unable to run the Matlab code (also I have to admit I'm not too well-versed in it). 
However, in addition to R, I also checked the result in SPSS and it yields the same result as JASP - see the attached screenshot. My hunch is that there is something happening under the hood of Matlab that I cannot pinpoint, or that there is a coding error in your Matlab code somewhere. I also noticed that your variable names are different in the Matlab code compared to the JASP file, so maybe there is something going on with that? 
Kind regards,
Johnny
<img width="704" alt="spsscheck" src="https://user-images.githubusercontent.com/15704203/37826192-c6b4b006-2e92-11e8-9b7e-9f1562f1b150.png">
